pkg/vm/emit/emit.go

```diff
@@ -32,20 +33,27 @@ func Bool(w *io.BinWriter, ok bool) {
 	Opcode(w, opcode.PUSHF)
 }
 
+func padRight(s int, buf []byte) []byte {
+	if len(buf) == s {
+		return buf
+	}
+	return buf[:s]
+}
+
 // Int emits a int type to the given buffer.
 func Int(w *io.BinWriter, i int64) {
 	switch {
 	case i == -1:
 		Opcode(w, opcode.PUSHM1)
-	case i == 0:
-		Opcode(w, opcode.PUSHF)
-	case i > 0 && i < 16:
+	case i >= 0 && i < 16:
 		val := opcode.Opcode(int(opcode.PUSH1) - 1 + int(i))
 		Opcode(w, val)
 	default:
-		bInt := big.NewInt(i)
-		val := IntToBytes(bInt)
-		Bytes(w, val)
+		buf := intToBytes(big.NewInt(i), make([]byte, 0, 32))
+		// l != 0 becase of switch
+		padSize := byte(8 - bits.LeadingZeros8(byte(len(buf)-1)))
+		Opcode(w, opcode.Opcode(padSize))
```

Comment:
Why not `w.WriteB()` here? `padSize` is not an opcode anyway.

Reply:
To not violate incapsulation. I know that `8, 16, 32, etc.` are sucessive opcodes starting with `0`, and that's all. May be `opcode.PUSHINT8 + padSize` will be more aproppriate.

Reply:
Yep, it would look a little better.